denonavr/denonavr.py

```diff
@@ -871,6 +879,28 @@ def _get_support_sound_mode(self):
         else:
             return self._get_support_sound_mode_avr()
 
+    def _update_tone_control(self):
+        """Update tone control related things."""
+        try:
+            root = self.exec_appcommand_post(attribute_list=['GetToneControl'])
+        except requests.exceptions.ConnectTimeout:
+            root = None
+
+        if root is None:
+            _LOGGER.error("Getting tone control failed.")
+            return False
+
+        try:
+            self._tone_control_adjust = bool(int(root[0].find('adjust').text))
+            self._bass = int(root[0].find('bassvalue').text)
+            self._bass_level = root[0].find('basslevel').text
+            self._treble = int(root[0].find('treblevalue').text)
+            self._treble_level = root[0].find('treblelevel').text
+        except (AttributeError, IndexError):
```

Comment:
On my receiver there are two kinds of responses
```xml
<?xml version="1.0" encoding="utf-8" ?>
<rx>
    <cmd>
        <status>0</status>
        <adjust></adjust>
        <basslevel></basslevel>
        <bassvalue></bassvalue>
        <treblelevel></treblelevel>
        <treblevalue></treblevalue>
    </cmd>
</rx>
```
and 
```xml
<?xml version="1.0" encoding="utf-8" ?>
<rx>
    <cmd>
        <status>1</status>
        <adjust>0</adjust>
        <basslevel>0dB</basslevel>
        <bassvalue>6</bassvalue>
        <treblelevel>0dB</treblelevel>
        <treblevalue>6</treblevalue>
    </cmd>
</rx>

The receiver returns the first result, when tone control is not active, like when Dynamic EQ is enabled. This causes a TypeError exception for the type conversions in those lines.

Please change two things:
1. Catch the TypeError exception
2. Check <status> before updating the variables in order to avoid spamming the log with error messages when tone control is not active

```

Reply:
> On my receiver there are two kinds of responses
> 
> ```
> <?xml version="1.0" encoding="utf-8" ?>
> <rx>
>     <cmd>
>         <status>0</status>
>         <adjust></adjust>
>         <basslevel></basslevel>
>         <bassvalue></bassvalue>
>         <treblelevel></treblelevel>
>         <treblevalue></treblevalue>
>     </cmd>
> </rx>
> ```
> 
> and
> 
> ```
> <?xml version="1.0" encoding="utf-8" ?>
> <rx>
>     <cmd>
>         <status>1</status>
>         <adjust>0</adjust>
>         <basslevel>0dB</basslevel>
>         <bassvalue>6</bassvalue>
>         <treblelevel>0dB</treblelevel>
>         <treblevalue>6</treblevalue>
>     </cmd>
> </rx>
> 
> The receiver returns the first result, when tone control is not active, like when Dynamic EQ is enabled. This causes a TypeError exception for the type conversions in those lines.
> 
> Please change two things:
> 1. Catch the TypeError exception
> 2. Check <status> before updating the variables in order to avoid spamming the log with error messages when tone control is not active
> ```

I have added `status` as attribute to check, if it is possible to interact with tone control.